portfolio/src/main/java/com/google/sps/servlets/MapMarkerServlet.java

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/map-marker")
+public class MapMarkerServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Query query = new Query("map-marker").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    ArrayList<Map<String, Double>> pairList = new ArrayList<Map<String, Double>>();
```

Comment:
Following along from Mike's suggestion, I think it would be better to have an actual class with named members, e.g.

---
class Location {
  public double latitude;
  public double longitude;
}

...

`ArrayList<Location> locations = new ArrayList<Location>();`

I think Gson can actually handle this case, which may blow your mind if you're used to C++, but Java has a concept called "reflection" that makes this work.

Reply:
Yeah that makes sense!

Reply:
I'm assuming this would work too, but I could make the class vars private and use getters/setters and the json format would be the same, right? I'm not sure why it would be different but I wanted to double check.

Reply:
Yes, your solution would also work.  Gson can use reflection to access private fields.

The public vs private is a tradeoff between added work of adding getters and setters vs protection of the class data members.  If you have a class (or in c++ a struct) which just holds plain old data without any functions then we usually forego creating setters/getters because you are just using the class as a way to group a bunch of data members.

Now if you had restrictions on the data member and for say you needed to ensure that it was always positive then here you would want to have a getter/setter so that in the setter you could have logic that guaranteed this.